### ISSUE TITLE
code engine create runtime need 'serverless:runtime' key or docker

### ISSUE THREAD

**Issue by gilv** [lithops-cloud/lithops#541]
there seems to be a bug in code_engine/config.py where the following code 
```
    if 'runtime' not in config_data['serverless']:
        if not DOCKER_PATH:
            raise Exception('docker command not found. Install docker or use '
                            'an already built runtime')
        if 'docker_user' not in config_data['code_engine']:
            cmd = "{} info".format(DOCKER_PATH)
            docker_user_info = sp.check_output(cmd, shell=True, encoding='UTF-8',
                                               stderr=sp.STDOUT)
            for line in docker_user_info.splitlines():
                if 'Username' in line:
                    _, useranme = line.strip().split(':')
                    config_data['code_engine']['docker_user'] = useranme.strip()
                    break

        if 'docker_user' not in config_data['code_engine']:
            raise Exception('You must provide "docker_user" param in config '
                            'or execute "docker login"')
```

requires either runtime key or docker_user. In case we built new runtime this code throw exception
``` 
Exception('You must provide "docker_user" param in config '
                            'or execute "docker login"')
```
                            
                            
which doesn't make sense when creating new runtimes. Even more, running  ``` docker login ``` doesn't make any change

**Comment by gilv** [comment]
@JosepSampe can we remove this or pass `runtime` from cli once we create new runtime?

**Comment by JosepSampe** [comment]
>> Even more, running docker login doesn't make any change

I think you have something wrong in your docker installation where the python scripts are unable to detect the docker command. I tested it in multiple windows, linux and macos hosts, and everywhere it detects the docker command, so this exception is never produced when building a runtime. is docker binary in your system path?

**Comment by gilv** [comment]
@JosepSampe at least one other person reported the same issue

**Comment by JosepSampe** [comment]
@gilv If you run this python code, what is the output?

```python
import shutil
print(shutil.which('docker'))
```

**Comment by gilv** [comment]
@JosepSampe `/usr/local/bin/docker`

**Comment by gilv** [comment]
@JosepSampe I just tested it again with most recent code.. the same problem for code-engine backend. 

```
Monte Carlo simulation for estimating PI spawing over 50 IBM Code Engine
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-6-b0b58d158f41> in <module>
      4 print("Monte Carlo simulation for estimating PI spawing over {} IBM Code Engine".format(MAP_INSTANCES))
      5 # obtain lithops executor
----> 6 lt = lithops.FunctionExecutor(mode="serverless", backend='code_engine')
      7 # execute the code
      8 t0 = time()

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/lithops/executors.py in __init__(self, mode, config, backend, storage, runtime, runtime_memory, rabbitmq_monitor, workers, remote_invoker, log_level)
     98             config_ow['lithops']['rabbitmq_monitor'] = rabbitmq_monitor
     99 
--> 100         self.config = default_config(copy.deepcopy(config), config_ow)
    101 
    102         self.executor_id = create_executor_id()

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/lithops/config.py in default_config(config_data, config_overwrite)
    169         logger.debug("Loading Serverless backend module: {}".format(sb))
    170         cb_config = importlib.import_module('lithops.serverless.backends.{}.config'.format(sb))
--> 171         cb_config.load_config(config_data)
    172 
    173         verify_runtime_name(config_data[constants.SERVERLESS]['runtime'])

~/.pyenv/versions/3.8.5/lib/python3.8/site-packages/lithops/serverless/backends/code_engine/config.py in load_config(config_data)
    174 
    175         if 'docker_user' not in config_data['code_engine']:
--> 176             raise Exception('You must provide "docker_user" param in config '
    177                             'or execute "docker login"')
    178 

Exception: You must provide "docker_user" param in config or execute "docker login"
```

**Comment by JosepSampe** [comment]
@gilv Interesting...

Can you show me the output of:

```python
import subprocess as sp
import shutil
DOCKER_PATH = shutil.which('docker')
cmd = "{} info".format(DOCKER_PATH)
docker_user_info = sp.check_output(cmd, shell=True, encoding='UTF-8', stderr=sp.STDOUT)
print(docker_user_info)
```

**Comment by gilv** [comment]
```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  scan: Docker Scan (Docker Inc., v0.5.0)

Server:
 Containers: 8
  Running: 0
  Paused: 0
  Stopped: 8
 Images: 807
 Server Version: 20.10.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.121-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 1.943GiB
 Name: docker-desktop
 ID: RZSY:JUND:BDYD:UGAR:D6UZ:HRVT:XVDC:ZD4P:ECOW:LVEJ:DSK6:UKPT
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 41
  Goroutines: 46
  System Time: 2021-02-03T05:26:18.1151692Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

@JosepSampe

**Comment by JosepSampe** [comment]
@gilv Thanks, I will fix it

**Comment by gilv** [comment]
@JosepSampe did you fixed it already? If yes, can you submit PR so we will include it with the next release?

**Comment by JosepSampe** [comment]
I have a tentative fix. I will submit a PR

**Comment by JosepSampe** [comment]
@gilv I already submitted a tentative fix #557. I tested it in one MAC OS and it works, please check if master branch works for you too
